packages/jest-mock/src/__tests__/index.test.ts

```diff
@@ -1073,6 +1074,62 @@ describe('moduleMocker', () => {
     });
   });
 
+  describe('withImplementation', () => {
+    it('sets an implementation which is available within the callback', () => {
+      const mock1 = jest.fn();
+      const mock2 = jest.fn();
+
+      const Module = jest.fn(() => ({someFn: mock1}));
+      const testFn = function () {
+        const m = new Module();
+        m.someFn();
+      };
+
+      Module.withImplementation(
+        () => ({someFn: mock2}),
+        () => {
+          testFn();
+          expect(mock2).toHaveBeenCalled();
+          expect(mock1).not.toHaveBeenCalled();
+        },
+      );
+
+      testFn();
+      expect(mock1).toHaveBeenCalled();
+
+      expect.assertions(3);
```

Comment:
Just wondering: should `expect.assertions` be added to the examples in docs as well? I mean, if users must use `expect` inside the callback, it would be good to show that in the example. Or?

Reply:
Since the callback function will always be executed immediately by `withImplementation` - no matter if it's async or not - I wouldn't think `expect.assertions` is necessary outside of jest's own test suite.

Reply:
Ah, right. Got it (;